javalib/src/main/scala/java/lang/Float.scala

```diff
@@ -293,11 +293,71 @@ object Float {
       zUp
   }
 
-  @inline private def multiplyBy10Pow(v: BigInteger, e: Int): BigInteger =
-    v.multiply(BigInteger.TEN.pow(e))
+  /** An implementation of big integer arithmetics that we need in the above method. */
+  private sealed abstract class BigIntImpl {
+    type Repr
+
+    def fromString(str: String): Repr
+
+    /** Creates a big integer from a `Long` that needs at most 53 bits (unsigned). */
+    def fromUnsignedLong53(x: scala.Long): Repr
+
+    def multiplyBy2Pow(v: Repr, e: Int): Repr
+    def multiplyBy10Pow(v: Repr, e: Int): Repr
+
+    def compare(x: Repr, y: Repr): Int
+  }
+
+  private object BigIntImpl {
+    /** Get the best available implementation of big integers for the given platform.
+     *
+     *  If JS bigint's are supported, use them. Otherwise fall back on
+     *  `java.math.BigInteger`.
+     *
+     *  We need a `linkTimeIf` here because the JS bigint implementation uses
+     *  the `**` operator, which does not link when `esVersion < ESVersion.ES2016`.
+     */
+    @inline def getImpl(): BigIntImpl = {
+      import scala.scalajs.LinkingInfo._
```

Comment:
Uber nit: Feel odd to import this here rather than on the top level. Feels like it is extremely unlikely that we'll collide with the name `linkTimeIf`.